airflow/api_fastapi/core_api/routes/public/variables.py

```diff
@@ -130,14 +132,24 @@ def patch_variable(
         raise HTTPException(
             status.HTTP_404_NOT_FOUND, f"The Variable with key: `{variable_key}` was not found"
         )
+
+    fields_to_update = patch_body.model_fields_set
     if update_mask:
+        fields_to_update = fields_to_update.intersection(update_mask)
         data = patch_body.model_dump(
-            include=set(update_mask) - non_update_fields, by_alias=True, exclude_none=True
+            include=fields_to_update - non_update_fields, by_alias=True, exclude_none=True
         )
     else:
+        try:
+            VariableBody(**patch_body.model_dump())
+        except ValidationError as e:
+            raise RequestValidationError(errors=e.errors())
         data = patch_body.model_dump(exclude=non_update_fields, by_alias=True, exclude_none=True)
+
     for key, val in data.items():
         setattr(variable, key, val)
+
+    session.commit()
```

Comment:
`commit` is automatically done on session close. I don't think we need to do that manually here.

Reply:
You can `flush` if you need auto-generated id/ or anything else from the db side.

Reply:
Okay, looks like both are only not required then, if only closing the session it will get committed.